### PR TITLE
Price liste client

### PR DESCRIPTION
**Pull request by lomamech**
- Update client side for Price List unit
- Add Test E2E for Price list Client side

@jniles  @DedrickEnc

### CODE REVIEW

File: client/test/e2e/price_list/price_list.spec.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+var FU = require('../shared/FormUtils');
+var helpers = require('../shared/helpers');
+var components = require('../shared/components');
+
+helpers.configure(chai);
+
+describe('Price List Module', function () {
+  'use strict';
+
+  var path = '#/inventory/price_list';
```
Comment (jniles) [inline client/test/e2e/price_list/price_list.spec.js]: This should be `#/prices`, I believe.

File: server/controllers/finance/priceList.js [hunk 2]
```diff
@@ -183,10 +194,8 @@ exports.create = function create(req, res, next) {
     'INSERT INTO price_list_item (uuid, inventory_uuid, price_list_uuid, ' +
     'label, value, is_percentage) VALUES ?;';
 
-
   // generate a UUID if not provided
-  data.uuid = data.uuid || uuid();
-
+  data.uuid = data.uuid || uuid.v4();
```
Comment (jniles) [inline server/controllers/finance/priceList.js:198]: :+1:

File: client/src/partials/price_list/pricelist.js [hunk 3]
```diff
@@ -1,262 +1,154 @@
...
-  'appstate', 'validate', 'uuid'
+  'PriceListService', '$window', '$translate', '$uibModal', 'InventoryService'
 ];
 
-function PriceListController($scope, $q, $translate, $window, connect, messenger, appstate, validate, uuid) {
-  var dependencies = {};
-  var enterprise;
+function PriceListController(PriceListService, $window, $translate, $uibModal, Inventory) {
+  var vm = this;
+  vm.session = {};
```
Comment (jniles) [inline client/src/partials/price_list/pricelist.js]: Do you use `vm.session`?  Why not just used `vm` instead?

File: client/src/partials/price_list/pricelist.html [hunk 4]
```diff
@@ -1,195 +1,202 @@
...
+                    <th>{{ "COLUMNS.PERCENT" | translate }}</th>
+                    <th>{{ "COLUMNS.INVENTORY" | translate }}</th>
+                    <th colspan="6">{{ "COLUMNS.ACTION" | translate }}</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  <tr ng-repeat="item in PriceListCtrl.pricelistItems | orderBy:'label'">
+                    <td> {{ item.label }} </td>
+                    <td> {{ item.value }} </td>
+                    <td> <span class="glyphicon" ng-class="{'glyphicon-ok' : !!item.is_percentage}"> </td>
```
Comment (jniles) [inline client/src/partials/price_list/pricelist.html]: You are missing the closing `</span>` tag here.

File: client/src/partials/price_list/pricelist.html [hunk 5]
```diff
@@ -1,195 +1,202 @@
...
+                    <td> {{ PriceListCtrl.getInventory(item.inventory_uuid) }} </td>
+                    <td colspan="6"><a class="btn btn-xs btn-default" id="remove_item_{{$index+1}}" ng-click="PriceListCtrl.removeItem(item)" ><i class="glyphicon glyphicon-trash white"></i></a></td>
```
Comment (jniles) [inline client/src/partials/price_list/pricelist.html]: What does the "white" class do?

File: client/src/partials/price_list/pricelist.html [hunk 6]
```diff
@@ -1,195 +1,202 @@
...
 
+        <div class="panel panel-primary" ng-switch-when="create">
+          <div class="panel-heading">{{ "PRICE_LIST.ADD" | translate }}</div>
           <div class="panel-body">
-            <form name="newPriceList" class="form-horizontal">
+            <form class="panel-body" name="CreateForm" ng-submit="PriceListCtrl.submit(CreateForm.$invalid)" novalidate>
+
+              <div class="form-group" ng-class="{ 'has-error' : CreateForm.$submitted && CreateForm.label.$invalid }">
+                <label class="control-label">{{ 'COLUMNS.DESIGNATION' | translate }}</label>
+                <input type="text" class="form-control" name="label" ng-model="PriceListCtrl.priceList.label" required>
```
Comment (jniles) [inline client/src/partials/price_list/pricelist.html:98]: Could you put some form validation messages here?  If you use `ng-messages`, they are very easy.

File: client/src/partials/price_list/pricelist.html [hunk 7]
```diff
@@ -1,195 +1,202 @@
...
+                    <th>{{ "COLUMNS.PERCENT" | translate }}</th>
+                    <th>{{ "COLUMNS.INVENTORY" | translate }}</th>
+                    <th colspan="6">{{ "COLUMNS.ACTION" | translate }}</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  <tr ng-repeat="item in PriceListCtrl.pricelistItems | orderBy:'label'">
+                    <td> {{ item.label }} </td>
+                    <td> {{ item.value }} </td>
+                    <td> <span class="glyphicon" ng-class="{'glyphicon-ok' : !!item.is_percentage}"> </td>
```
Comment (jniles) [inline client/src/partials/price_list/pricelist.html]: This is also missing a closing `</span>` tag.

File: client/src/partials/price_list/pricelist.html [hunk 8]
```diff
@@ -1,195 +1,202 @@
...
+        <div class="panel panel-default" style="overflow: auto; max-height: 500px;">
+          <div class="panel-heading">
+            {{ "PRICE_LIST.LISTS" | translate }}
+          </div>
+
+          <table class="table table-condensed">
+            <thead>
+              <tr>
+                <th>{{ "COLUMNS.DESCRIPTION" | translate }}</th>
+                <th colspan="3" style="width: 20%;">{{ "COLUMNS.ACTION" | translate }}</th>
```
Comment (jniles) [inline client/src/partials/price_list/pricelist.html]: All these colspans do not line up.  I haven't looked at the rendered page, but it looks like you have four columns here,
three in the `<tbody>` and only two if there is no data.  Can you verify that this renders properly?

File: client/src/partials/price_list/modal.html [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    </div>
+
+    <div class="form-group" ng-class="{ 'has-error' : ModalForm.$submitted && ModalForm.value.$invalid }">
+      <label class="control-label">{{ 'COLUMNS.VALUE' | translate }}</label>
+      <input type="number" class="form-control" name="value" ng-model="ModalCtrl.data.value" required>
+    </div>
+
+    <div class="form-group" ng-class="{ 'has-error' : ModalForm.$submitted && ModalForm.is_percentage.$invalid }">
+      <label class="control-label">{{ 'COLUMNS.IS_PERCENT' | translate }}</label>
+      <input type="checkbox" class="form-control" name="is_percentage" id="is_percentage" ng-true-value="1" ng-false-value="0" ng-model="ModalCtrl.data.is_percentage">
```
Comment (jniles) [inline client/src/partials/price_list/modal.html]: The proper way to do checkboxes in bootstrap is shown [here](http://getbootstrap.com/css/#forms-example).  The code should look something like:

``` html
  <div class="checkbox">
    <label>
      <input type="checkbox"> Check me out
    </label>
  </div>
```

File: client/src/partials/price_list/modal.html [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    </div>
+
+    <div class="form-group" ng-class="{ 'has-error' : ModalForm.$submitted && ModalForm.inventory_uuid.$invalid }">
+      <label class="col-md-3 control-label">{{ "COLUMNS.INVENTORY" | translate }}</label>
```
Comment (jniles) [inline client/src/partials/price_list/modal.html]: Why does this label have `col-md-3` as a class?

**Comment by jniles** [comment]
@lomamech, I've given some initial feedback on the code.  We will see how this works on Monday.

**Comment by lomamech** [comment]
@jniles  
Update and Improve of code for Price List

**Comment by DedrickEnc** [comment]
A ce que je sache, ce code etait deja revu, et tu as apportE des modifications concernant la revision donc je le melange car je ne vois pas grand chose a completer.

**Pull request merged**